hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/test/java/com/google/cloud/bigtable/hbase/replication/adapters/ApproximatingIncompatibleMutationAdapterTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigtable.hbase.replication.adapters;
+
+import static com.google.cloud.bigtable.hbase.replication.adapters.ApproximatingIncompatibleMutationAdapter.DELETE_FAMILY_WRITE_THRESHOLD_KEY;
+import static org.apache.hadoop.hbase.HConstants.LATEST_TIMESTAMP;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.bigtable.hbase.replication.metrics.MetricsExporter;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.KeyValue;
+import org.apache.hadoop.hbase.client.Connection;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@RunWith(JUnit4.class)
+public class ApproximatingIncompatibleMutationAdapterTest {
+
+  private static final byte[] rowKey = "rowKey".getBytes(StandardCharsets.UTF_8);
+  private static final byte[] cf = "family".getBytes(StandardCharsets.UTF_8);
+  private static final byte[] qual = "qual".getBytes(StandardCharsets.UTF_8);
+  private static final byte[] val = "value".getBytes(StandardCharsets.UTF_8);
+
+  @Rule public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  Configuration conf;
+
+  @Mock Connection connection;
+
+  @Mock MetricsExporter metricsExporter;
+
+  @Mock BigtableWALEntry mockWalEntry;
+
+  ApproximatingIncompatibleMutationAdapter incompatibleMutationAdapter;
+
+  @Before
+  public void setUp() throws Exception {
+    conf = HBaseConfiguration.create();
```

Comment:
Does this need to be a n HBaseConfiguration and do you want it to search for hbase-site.xml? if not then this should be `new Configuration(false)`